interfaces/builtin/desktop_test.go

```diff
@@ -91,6 +94,40 @@ func (s *DesktopInterfaceSuite) TestAppArmorSpec(c *C) {
 	c.Assert(spec.SecurityTags(), HasLen, 0)
 }
 
+func (s *DesktopInterfaceSuite) TestMountSpec(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	// On all-snaps systems, no mount entries are added
+	spec := &mount.Specification{}
+	c.Assert(spec.AddConnectedPlug(s.iface, s.plug, nil, s.coreSlot, nil), IsNil)
+	c.Check(spec.MountEntries(), HasLen, 0)
+
+	// On classic systems, a number of font related directories
+	// are bind mounted from the host system if they exist.
+	release.OnClassic = true
+	spec = &mount.Specification{}
+	c.Assert(spec.AddConnectedPlug(s.iface, s.plug, nil, s.coreSlot, nil), IsNil)
+	expectedMountPoints := []string{
+		"/usr/share/fonts",
+		"/usr/local/share/fonts",
+		"/var/cache/fontconfig",
+	}
+	expected := 0
+	for _, dir := range expectedMountPoints {
+		if osutil.IsDirectory(dir) {
+			expected += 1
+		}
+	}
+	entries := spec.MountEntries()
+	c.Check(entries, HasLen, expected)
+	for _, entry := range entries {
+		c.Check(expectedMountPoints, testutil.Contains, entry.Dir)
+		c.Check(entry.Name, Equals, "/var/lib/snapd/hostfs"+entry.Dir)
+		c.Check(entry.Options, DeepEquals, []string{"bind", "ro"})
```

Comment:
This will *probably* need corresponding snap-confine apparmor profile updates. While I recall adding one for `/usr/share/fonts` well mid last year I'd like to check if all of those are present.

Reply:
I didn't notice any problems in my manual testing.  I wonder if `snap-update-ns` got in first so I never hit that code path in `snap-confine`?